editor/src/components/navigator/left-pane.tsx

```diff
@@ -763,6 +768,259 @@ const GithubSpinner = () => {
   )
 }
 
+interface RepositoryRowProps extends RepositoryEntry {
+  importPermitted: boolean
+}
+
+const RepositoryRow = (props: RepositoryRowProps) => {
+  const colorTheme = useColorTheme()
+
+  const dispatch = useEditorState((store) => store.dispatch, 'RepositoryRow dispatch')
+
+  const projectID = useEditorState((store) => store.editor.id, 'RepositoryRow projectID')
+
+  const importRepository = React.useCallback(() => {
+    const parsedTargetRepository = parseGithubProjectString(props.fullName)
+    if (parsedTargetRepository == null || props.defaultBranch == null) {
+      dispatch(
+        [
+          showToast(
+            notice(
+              `Error when attempting to import a repository with repo: ${props.fullName}`,
+              'ERROR',
+            ),
+          ),
+        ],
+        'everyone',
+      )
+    } else {
+      void getBranchContent(
+        dispatch,
+        parsedTargetRepository,
+        forceNotNull('Should have a project ID.', projectID),
+        props.defaultBranch,
+      )
+    }
+  }, [dispatch, projectID, props.fullName, props.defaultBranch])
+
+  return (
+    <div
+      style={{
+        minHeight: 40,
+        display: 'flex',
+        flexDirection: 'row',
+        gap: 8,
+        alignItems: 'center',
+        paddingLeft: 8,
+        paddingRight: 8,
+        paddingBottom: 8,
+        paddingTop: 8,
+        borderBottom: '1px solid #ccc',
+      }}
+    >
+      <div
+        style={{
+          borderRadius: '50%',
+          width: 20,
+          height: 20,
+          border: '1px solid #ccc',
+          backgroundImage: `url(${props.avatarUrl})`,
+          backgroundPosition: 'center',
+          backgroundSize: 'cover',
+          backgroundRepeat: 'no-repeat',
+        }}
+      />
+      <div style={{ flexGrow: 1 }}>
+        <span style={{ fontWeight: 600, textOverflow: 'ellipsis' }}>
+          {props.name ?? props.fullName}
+        </span>{' '}
+        <br />
+        <span style={{ opacity: 0.5 }}>
+          {props.private ? 'private' : 'public'}
+          {props.updatedAt == null ? null : (
+            <>
+              {' · '}
+              <TimeAgo date={props.updatedAt} />
+            </>
+          )}
+        </span>
+      </div>
+      <button
+        style={{
+          fontSize: 11,
+          background: colorTheme.buttonBackground.value,
+          boxShadow: 'none',
+          border: 'none',
+          height: 22,
+          color: props.importPermitted
+            ? colorTheme.inlineButtonColor.value
+            : colorTheme.inlineButtonColor.shade(50).value,
+          borderRadius: 2,
+          cursor: 'pointer',
+        }}
+        disabled={!props.importPermitted}
+        onMouseUp={importRepository}
+      >
+        Import
```

Comment:
One last thing: we should make sure that this is:
1. disabled while there are ongoing github operations
2. showing the spinner during the import like the other buttons

This requires some minor changes, here's a suggestion:

In `left-pane.tsx`:
```ts
const RepositoryRow = (props: RepositoryRowProps) => {
  // ...
  const githubOperations = useEditorState(
    (store) => store.editor.githubOperations,
    'Github operations',
  )

  const githubWorking = React.useMemo(() => githubOperations.length > 0, [githubOperations])

  const [importing, setImporting] = React.useState(false)

  React.useEffect(() => {
    if (
      !githubOperations.some(
        (op) => op.name === 'loadBranch' && githubRepoName(op.repository) === props.fullName,
      )
    ) {
      setImporting(false)
    }
  }, [githubOperations, props.fullName])

  // ...

  const importRepository = React.useCallback(() => {
    const parsedTargetRepository = parseGithubProjectString(props.fullName)
    if (parsedTargetRepository == null || props.defaultBranch == null) {
      // ...
    } else {
      setImporting(true)
      void getBranchContent(
      // ...
    }

    // ...
    <button
       // ...
        disabled={!props.importPermitted || githubWorking}
       // ...
      >
        {importing ? <GithubSpinner /> : 'Import'}
      </Button>
```

In `editor-state.ts`:
```ts
export type GithubOperation =
  | { name: 'commish' }
  | { name: 'listBranches' }
  | { name: 'loadBranch'; repository: GithubRepo; branchName: string }
  | { name: 'loadRepositories' }
```

In `github.ts`:
```ts
export async function getBranchContent(
  dispatch: EditorDispatch,
  githubRepo: GithubRepo,
  projectID: string,
  branchName: string,
): Promise<void> {
  const operation: GithubOperation = {
    name: 'loadBranch',
    repository: githubRepo,
    branchName: branchName,
  }
  // ...
```

Reply:
Fixed in 19dd7446e.

Reply:
it'd be good to add the `importing` logic as well, or all the import buttons will be spinning while github is working (see the example snippet above for my naive take on it)

<img width="285" alt="Screen Shot 2022-11-01 at 12 48 43 PM" src="https://user-images.githubusercontent.com/1081051/199226357-521f85d9-33f1-4240-b134-95291f4493a4.png">